src/main/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java

```diff
@@ -139,37 +168,50 @@ public static Tool deserialize(final IFactoryController controller, final Friend
         return new Tool(type, minLevel, maxLevel, result);
     }
 
-    /**
-     * Returns the tool class that is requested.
-     *
-     * @return The tool class that is requested.
-     */
-    @NotNull
-    public IToolType getToolClass()
+    @Override
+    public boolean matches(@NotNull final ItemStack stack)
     {
-        return toolClass;
+        if (ItemStackUtils.isEmpty(stack))
+        {
+            return false;
+        }
+
+        if (stack.getCount() < 1)
+        {
+            return false;
+        }
+
+        for (RegistryObject<ToolTypeEntry> toolType : ModToolTypes.toolTypes)
+        {
+            if (!getToolClass().equals(toolType.get()))
```

Comment:
can't we == those?

Reply:
It's an interface, should use equals

Reply:
Aahh, I thought we were comparing two types.
But that opens up the question, why isn't getToolClass returning the same type of type?

Reply:
Yeah actually, what is this entire loop good for anyway? It only contains a single IToolType which is checked against all tool types, so what is the point here?

Reply:
Fixed. This doesn't actually need to loop now with the new logic. And also should work for paxels, etc.